vignettes/templates.Rmd

```diff
@@ -0,0 +1,350 @@
+           "<0.001" ~ "<0.001",
+           TRUE ~ frmt("x.xxx", missing = "--")
+           )
+
+n_pct_frmt <- frmt_combine(
+          "{n} {pct}",
+          n = frmt("XXX"),
+          pct = frmt_when(
+            "==100" ~ "",
+            "==0" ~ "",
+            TRUE ~ frmt("(xx.x %)")
+            )
+          )
+
+## frmts as functions
+
+int_frmt_func <- function(ints = 2){
+  str_exp <- paste0("[", paste0(rep("X", ints), collapse = ""), "]")
+  frmt(str_exp)
+}
+
+
+```
+ 
+Now that there are defined frmts, we can use them in our tfrmt to generate our table too.
+
+
+```{r}
+
+tfrmt(
+    group = AEBODSYS,
+    label = AETERM,
+    param = param,
+    column = treatment,
+    value = value,
+    
+    body_plan = body_plan(
+      frmt_structure(
+        group_val = ".default",
+        label_val = ".default",
+        AEs = int_frmt_func(3)
+        ),
+      
+      frmt_structure(
+        group_val = ".default",
+        label_val = ".default",
+        n_pct_frmt
+        ),
+      
+      frmt_structure(
+         group_val = ".default", 
+         label_val = ".default",
+         pval = pval_frmt
+      )
+    ),
+    
+    ## remove order columns from final table
+    col_plan = col_plan(
+      -ord1,-ord2,
+      span_structure(
+        "Xanomeline High Dose (N=84)",
+        `n (%)` = `Xanomeline High Dose-n_pct` ,
+        `[AEs]` = `Xanomeline High Dose-AEs`
+      ),
+      span_structure(
+        "Xanomeline Low Dose (N=84)",
+        `n (%)` = `Xanomeline Low Dose-n_pct` ,
+        `[AEs]` = `Xanomeline Low Dose-AEs`
+      ),
+      span_structure("Placebo (N=86)",
+                     `n (%)` = `Placebo-n_pct` ,
+                     `[AEs]` = `Placebo-AEs`),
+      span_structure(
+        "Fisher's Exact p-values",
+        `Placebo vs. Low Dose` = `fisher_pval-p_low` ,
+        `Placebo vs. High Dose` = `fisher_pval-p_high`
+      )
+    )
+    
+  ) %>% 
+  print_to_gt(ae_data2) %>% 
+  tab_options(
+    container.width = 1000
+  )
+
+```
+
+
+## Best Practices
+
+Here we list some of the ideas about best practices when it comes to defining tfrmt templates for use across organization. This comes from experience writing and using `tfrmt` and has the potential to change. 
+
+Creating a template tfrmt is as much about the table you are trying to standardize as what should be considered to change across tables. These functions should serve to provide a way to interact with and create tfrmt's for standard tables in such a way that customization for a specific table's `tfrmt` is simplified. However, as with normal function development, this should be balanced with limiting the number of arguments necessary to pass and get a functional tfrmt. For example, `tfrmt_sigdig` accepts a minimal set of arguments, including a data.frame or tibble, to define a complex body_plan. Simplifying the way users may enter this complex set of instructions without having to manually type it is part of its strength.
```

Comment:
```suggestion
When creating a template tfrmt, it is helpful to capture standard aspects of the data/display, rather than aspects that are likely to change across tables or studies. These functions should serve to provide a way to interact with and create tfrmt's for standard tables in such a way that customization for a specific table's `tfrmt` is simplified. However, as with normal function development, this should be balanced with limiting the number of arguments necessary to pass and get a functional tfrmt. For example, `tfrmt_sigdig` accepts a minimal set of arguments, including a data.frame or tibble, to define a complex body_plan. Simplifying the way users may enter this complex set of instructions without having to manually type it is part of its strength.
```